src/luascript.cpp

```diff
@@ -10824,6 +10830,45 @@ int LuaScriptInterface::luaPlayerGetIdleTime(lua_State* L)
 	return 1;
 }
 
+// OfflinePlayer
+int LuaScriptInterface::luaOfflinePlayerCreate(lua_State* L)
```

Comment:
1) If you pass a value other than a number or string then a null Player is returned to Lua, (I don't know how that behaves, but it shouldn't be like that).
2) You can get players that are online, it doesn't make sense because this object is supposed to be for offline players.
3) Use camelCase format.

Example:
```c++
// OfflinePlayer
int LuaScriptInterface::luaOfflinePlayerCreate(lua_State* L)
{
	// OfflinePlayer(guid or name)
	auto offlinePlayerPtr = std::make_unique<Player>(nullptr);
	if (const uint32_t guid = getNumber<uint32_t>(L, 2)) {
		if (g_game.getPlayerByGUID(guid) || !IOLoginData::loadPlayerById(offlinePlayerPtr.get(), guid)) {
			lua_pushnil(L);
			return 1;
		}
	} else if (const std::string& name = getString(L, 2); !name.empty()) {
		if (g_game.getPlayerByName(name) || !IOLoginData::loadPlayerByName(offlinePlayerPtr.get(), name)) {
			lua_pushnil(L);
			return 1;
		}
	}

	if (offlinePlayerPtr) {
		pushUserdata<Player>(L, offlinePlayerPtr.release());
		setMetatable(L, -1, "OfflinePlayer");
	} else {
		lua_pushnil(L);
	}
	return 1;
}
```